File: java/ZetCode/Swing/01 - First Programs/src/Tooltip.java
import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JPanel;
import javax.swing.SwingUtilities;


public class Tooltip extends JFrame
{
    public Tooltip()
    {
        initUi();
    }

    public final void initUi()
    {
        JPanel panel = new JPanel();
        this.getContentPane().add(panel);
        panel.setLayout(null);
        panel.setToolTipText("A JPanel widget");

        JButton button = new JButton("Button");
        button.setBounds(100, 60, 100, 30);
        button.setToolTipText("A JButton widget");
        panel.add(button);

        this.setTitle("Tooltip");
        this.setSize(300, 200);
        this.setLocationRelativeTo(null);
        this.setDefaultCloseOperation(EXIT_ON_CLOSE);
    }

    public static void main(String[] args)
    {
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                Tooltip ex = new Tooltip();
                ex.setVisible(true);
            }
        });
    }
}

File: python/Zetcode/Jython Swing Tutorial/01 - Introduction/tooltip.py
#!/usr/bin/env jython
# -*- encoding: utf-8 -*-

from javax.swing import JFrame
from javax.swing import JPanel
from javax.swing import JButton

class Example(JFrame):
    
    def __init__(self):
        JFrame.__init__(self)
        self.initUI()
        
    def initUI(self):
        panel = JPanel()
        self.getContentPane().add(panel)
        
        panel.setLayout(None)
        panel.setToolTipText("A panel container")
        
        button = JButton("Button")
        button.setBounds(100, 60, 100, 30)
        button.setToolTipText("A button component")
        
        panel.add(button)
        
        self.setTitle("Tooltips")
        self.setSize(300, 200)
        self.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)
        self.setLocationRelativeTo(None)
        self.setVisible(True)
        
        
if __name__ == "__main__":
    Example()